### PR TITLE
[connectivity] Add the ability to detect metered networks and whether Data Saver is active in real time on Android.

### PR DESCRIPTION
**Pull request by konstantintuev**
## Description

Added the ability to see if data restrictions on Android such as Data Saver or connection to Metered Network are active.

Defaults to none on other platform other than Android.

## Related Issues

No related issues.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/connectivity/lib/connectivity.dart [hunk 1]
```diff
@@ -14,6 +14,24 @@ import 'package:meta/meta.dart';
 /// None: Device not connected to any network
 enum ConnectivityResult { wifi, mobile, none }
 
+/// Metered: Device connected via metered Wi-Fi (Android only)
```
Comment (cyanglaz) [inline packages/connectivity/lib/connectivity.dart]: put docs on each option above the option itself.
e.g
```
/// Device connected via metered Wi-Fi (Android only)
metered
```
Also, add doc to explain this enum.

File: packages/connectivity/lib/connectivity.dart [hunk 2]
```diff
@@ -14,6 +14,24 @@ import 'package:meta/meta.dart';
...
+/// WhitelistedBackgroundData: DataSaver is active but the app is whitelisted can use background data (Android only)
+/// None: No DataSaver is active, default value on iOS
+enum DataSaving {
+  metered,
+  blockedBackgroundData,
+  whitelistedBackgroundData,
+  none
+}
+
+class NetworkInfo {
```
Comment (cyanglaz) [inline packages/connectivity/lib/connectivity.dart]: doc

File: packages/connectivity/lib/connectivity.dart [hunk 3]
```diff
@@ -14,6 +14,24 @@ import 'package:meta/meta.dart';
...
+  NetworkInfo(this.connectivityResult, this.dataSaving);
+
+  final ConnectivityResult connectivityResult;
```
Comment (cyanglaz) [inline packages/connectivity/lib/connectivity.dart]: doc on public members

File: packages/connectivity/lib/connectivity.dart [hunk 4]
```diff
@@ -95,14 +115,38 @@ class Connectivity {
...
+      case 'metered':
+        dataSaving = DataSaving.metered;
+        break;
+      case 'blockedBackgroundData':
+        dataSaving = DataSaving.blockedBackgroundData;
+        break;
+      case 'whitelistedBackgroundData':
+        dataSaving = DataSaving.whitelistedBackgroundData;
+        break;
+    }
```
Comment (cyanglaz) [inline packages/connectivity/lib/connectivity.dart]: add default case

**Comment by konstantintuev** [comment]
Thank you for the review, I am quite busy right now but as soon as I have a bit more free time I will update and push it.

**Pull request closed without merging**